actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -77,12 +87,15 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(record_or_options, additional_options = {})
-      if record_or_options.is_a? Hash
-        options = record_or_options
+    def fresh_when(record_or_collection_or_options, additional_options = {})
+      if record_or_collection_or_options.is_a? Hash
+        options = record_or_collection_or_options
         options.assert_valid_keys(:etag, :last_modified, :public, :template)
+      elsif record_or_collection_or_options.respond_to? :maximum
+        records  = record_or_collection_or_options
+        options = { etag: records, last_modified: records.try(:maximum, :updated_at) }.merge!(additional_options)
```

Comment:
If a developer overrides a system method and changes the method signature, they are just asking for trouble :/

But I guess you can keep `try` if you don't want to deal with it.